completed project

David-Nakeeran opened the pull request.

David-Nakeeran committed:
completed project

David-Nakeeran committed:
updated

David-Nakeeran committed:
Uncommented code

David-Nakeeran committed:
formatting

David-Nakeeran committed:
added placeholder values

David-Nakeeran committed:
updated password

chrisjamiecarter reviewed (approved):
Hey @David-Nakeeran 👋,

Excellent work on your **Phonebook** project submission 🎉!

I have performed a peer review. Review/ignore any comments as you wish.

***

🟢 **Requirements**

⭐ You have fulfilled all of the project requirements!

🟢 **README File**

⭐ Thank you, it helped me quickly set up and run your app!

❗ However, you did not call out that the user has to run migrations first (see next points to help with this).

🟢 **Email Functionality**

⭐ TOP!!!

🟡 **Code Style**

⭐ I didn't add any in-code comments as I thought it was a great project, the only nit-picky thing I would say is to look at your code and think about spacing. i.e. too much empty voids of whitespace; and the reverse, too much condensely packed code.

🟡 **Automated Migrations**

💭 FOR TESTING ONLY. You could add the below to your code to help with code reviewer set up. This is not recommended in a real-life full-scale app due to start-up and scaling concerns, but make sure if you do this in future, to call it out in your README.

❗ But, if you do not do this. Make sure you call out in a prerequisite step in your README that the user needs to create database / apply Migrations. Otherwise the app will throw an exception if the database doesn't exist.

![image](https://github.com/user-attachments/assets/7143f275-9a7f-44c3-b147-2c4b1bb836c7)

🟠 **appsettings Configuration**

💡 You need to remember to include your appsettings in your build/output location, as I got an exception thrown on initial start-up.

❌ 
![image](https://github.com/user-attachments/assets/497c2d52-b55c-43f0-a010-5615ff49f461)

✔️ 
![image](https://github.com/user-attachments/assets/2770c98a-3153-4580-ae6a-c07c1f373db9)

OR ✔️ 
![image](https://github.com/user-attachments/assets/f1b70e30-2585-4e77-bcb2-d8ce766541c4)

🟠 **User Experience**

❗ Ugly Enum names... 

👨‍🏫 Look in enum display attributes to enhance this for better reading.

![image](https://github.com/user-attachments/assets/34faef41-5f44-4820-9dbb-3818abd6b1d0)

🟠 **User Experience**

❓ Arrow keys to navigate? Lies! I had to enter an ID 😆 

👨‍🏫 Probably a hangover text from the main menu, but you could easily plugged in Spectre to enhance your user inputs.

![image](https://github.com/user-attachments/assets/7a8fec1e-3b01-46fb-bf7b-50138ff1ae42)

***

Feel free to reach out if you have any questions or if you'd like to collaborate further on any of these points 🆘 .

I will go ahead and mark as approved, keep up the excellent work on the next projects! 😊

Best regards,
@chrisjamiecarter 👍

David-Nakeeran commented:
Thank you for the code review Chris. 

I have updated my readme informing users to run migrations first.
Made sure that appsettings now gets copied to ouput directory.
Also used the display attribute for my enums.
Added in the forgotten console clear for good measure!

Many thanks again for the code review :)
